### ISSUE TITLE
Typescript client - FileResponse type not generated

### ISSUE THREAD

**Issue by ElGusanito** [RSuter/NSwag#2034]
Generated ts client references a type (FileResponse) that isn't defined:

protected processGetBookingRequestMi(response: HttpResponseBase): Observable<**FileResponse**> ...

**Swagger.JSON:**

responses:{200:
		{description:"Success",schema:
						{$ref:"#/definitions/FileContentResult"
		}
	  }

**API:**

        [ProducesResponseType(typeof(FileContentResult),200)]
        [Produces(contentType: "application/octet-stream")]
        public async Task<ActionResult> GetBookingRequestMi()
        {...

I have tried various permutations e.g. returning FileStreamResult, File, Byte[] - all have the same end result & I get a client referencing FileResponse which is undefined - although the proxy FileContentResult class is generated as expected.

Have also tried doing the same using the SwaggerResponse attribute but that doesn't seem to generate any schema at all & I can't generate the client

**Comment by RSuter** [comment]
try `typeof(byte[])`

**Comment by ElGusanito** [comment]
Same thing - still get the undefined FileResponse reference.

Changing to byte[] gives the follwing swagger.json:

responses":{"200":{"description":"Success","schema":{"format":"byte","type":"string"}}}

**Comment by RSuter** [comment]
Sorry, try one of these

![image](https://user-images.githubusercontent.com/2603405/54712307-3d53a380-4b4c-11e9-933b-e9ba2bb03653.png)

**Comment by RSuter** [comment]
The response must be

    responses":{"200":{"description":"Success","schema":{"type":"file"}}}

in Swagger 2.0 in order that the file response handling kicks in

**Comment by ElGusanito** [comment]
You mean like [ProducesResponseType(typeof(FileResult),200)] ?

That still does exactly the same thing & I get FileResponse

**Comment by ElGusanito** [comment]
responses":{"200":{"description":"Success","schema":{"type":"file"}}}

I just cannot generate this no matter what

**Comment by RSuter** [comment]
responses":{"200":{"description":"Success","schema":{"type":"file"}}}

Looks good.

Can you provide a complete swagger.json?

**Comment by ElGusanito** [comment]
{"operationId":"GetXXXXX","consumes":[],"produces":["application/octet-stream"],"parameters":[],"responses":{"200":{"description":"Success","schema":{"$ref":"#/definitions/FileContentResult"}}}}

**Comment by ElGusanito** [comment]
I can't produce this - responses":{"200":{"description":"Success","schema":{"type":"file"}}}

**Comment by ElGusanito** [comment]
Also, according to the documentation, removing the ResponseType attribute and changing the return type to IActionResult should cause it to default to file in the swagger schema.  So I changed my method signature:

public async Task<IActionResult> GetBookingRequestMi()

But in the swagger.json,  there isn't even a schema defined:

responses":{"200":{"description":"Success"}}

**Comment by RSuter** [comment]
Looks good to me:

![image](https://user-images.githubusercontent.com/2603405/54716913-34b49a80-4b57-11e9-8ab7-ad3034734397.png)

**Comment by RSuter** [comment]
With AddOpenApiDocument() (openapi 3) this gives me:

![image](https://user-images.githubusercontent.com/2603405/54716961-59107700-4b57-11e9-8662-8a5319f87adc.png)

Same with `[ProducesResponseType(typeof(ActionResult), 200)]`

**Comment by ElGusanito** [comment]
I'm using Swashbuckle.AspNetCore v 4.0.1  to generate the swagger.json - it seems that this produces a different format to what you are getting.

**Comment by ElGusanito** [comment]
Solved by swapping out Swashbuckle.AspNetCore with NSwag.AspNetCore

**Comment by older** [comment]
This also happens to me after I updated from 12.1.0 to 12.2.1. TypeScript client generated for this controller method:
```C#
        [HttpGet("file")]
        public IActionResult GetFile()
        {
            return File(new byte[0], "application/pdf");
        }
```
contains definition for `FileResponse` type when generated with v12.1.0 but this definition is missing when generated with v12.2.1.
Here's test project to reproduce this: [TestNSwag.zip](https://github.com/RicoSuter/NSwag/files/3107014/TestNSwag.zip)

**Comment by RicoSuter** [comment]
Fixed in v12.2.2
